### ISSUE TITLE
Indeterminate Checkbox - Incorrect message announced to assistive tech

### ISSUE THREAD

**Issue by orion-cengage** [cengage/react-magma#1231]
CENG_3302

Docs: https://react-magma.cengage.com/version/4.3.0/api/indeterminate/

Success Criteria: https://www.w3.org/WAI/WCAG21/Understanding/name-role-value

Within the Indeterminate checkbox, when the state of a checkbox is `indeterminate` that message is not announce to assistive technology. The message that is announced is not checked.

Use aria-checked="mixed" for an indeterminate checkbox when some of, but not all of, it's children are selected. A detailed example of this can be found here: https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/examples/checkbox-mixed/

**Comment by silvalaura** [comment]
We use indeterminate checkboxes inside multiple components, so let's confirm this doesn't affect those negatively. Example: TreeView and DataGrid
